app/kuma-cp/cmd/run.go

```diff
@@ -104,6 +104,10 @@ func newRunCmdWithOpts(opts runCmdOpts) *cobra.Command {
 					runLog.Error(err, "unable to set up Monitoring Assignment server")
 					return err
 				}
+				if err := kds_server.SetupServer(rt, []model.ResourceType{mesh.DataplaneType}); err != nil {
```

Comment:
I think there should be some kind of `kds_server.SetupServerForLocal(rt)` or so. IMHO `run.go` should not pick resources here